### PR TITLE
Enable fallback synthetic source by default

### LINKED ISSUE #106460: Support synthetic source for all field types
### Description

Some field types are still missing support for synthetic source. We would like to implement the missing functionality adopting a strategy that allows us to use a stored field behind the scenes. The idea is to use the stored field later on to support reconstructing the original field value instead of reconstructing it from doc values.

The ones we expect to be used more frequently for logging and that we should prioritise when it comes to implementing support are:
- [x] binary #107549
- [x] object (`enabled: false`) #108051
- [x] nested #108818
- [x] long_range #107081
- [x] double_range #107081
- [x] date_range #107081
- [x] ip_range #107081
- [x] murmur3
- [x] annotated-text #107735

For fields that are less frequently used we will implement a less storage efficient fallback implementation of synthetic source. This implementation will cover all fields that are not currently supported and unblock their usage in logs index mode. Based on usage of such fields, more efficient implementation can be added.
- [ ] join
- [ ] completion
- [x] search_as_you_type #108222, #108878
- [x] token_count #109044
- [ ] sparse_vector
- [ ] rank_feature
- [ ] rank_features
- [x] geo_shape #108881
- [ ] point #109312
- [ ] shape #109312
- [ ] percolator

We would like to enable this feature by default if synthetic source is enable. This will enable usage of these field types for both logs and TSDB index mode.
In general support should be provided by making the `stored` property default to `true` when synthetic source is enabled. This implies we need to add the `stored` property even for fields that are missing it.

Once support is implemented please check the box next to the corresponding type and link the PR next to it.

### PR DESCRIPTION
**Pull request by lkts**
This PR enables fallback synthetic source as a default for all fields that don't implement synthetic source support. It also adds basic tests for fields that now support synthetic source with that above change.

Closes #106460.

### CODE REVIEW

**Comment by lkts** [comment]
@elasticmachine update branch

**Review by kkrik-es — APPROVED** [review]
Let's `test-full-bwc` to play it safe.

File: server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java [hunk 1]
```diff
@@ -445,25 +445,24 @@ public Map<String, NamedAnalyzer> indexAnalyzers() {
...
         NATIVE,
-        FALLBACK,
-        NOT_SUPPORTED
+        FALLBACK
     }
 
     /**
      * Specifies the mode of synthetic source support by the mapper.
+     * Field mappers must override this method if they provide
+     * a custom implementation of {@link #syntheticFieldLoader()}
```
Comment (kkrik-es) [inline server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java:460]: Consider adding a note there hinting why they should do so.

**Review by martijnvg — APPROVED** [review]
lgtm

**Pull request merged**